components/resolved_ts/src/resolver.rs

```diff
@@ -568,8 +568,17 @@ impl Resolver {
         self.read_progress.as_ref()
     }
 
-    // Return the transaction with the smallest min_commit_ts. When min_commit_ts
-    // is unknown, use start_ts instead.
+    // Returns the minimum possible (commit_ts - 1), based on the knowledge we have,
+    // i.e. from all locks currently being tracked.
+    // The function is to provide an upper bound of resolved-ts. By definition
+    // resolved-ts must be strictly larger than a future commit_ts.
```

Comment:
```suggestion
    // resolved-ts must be strictly smaller than a future commit_ts.
```

If I understand correctly, it's `resolved-ts < commit-ts`.